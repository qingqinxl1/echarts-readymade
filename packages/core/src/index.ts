import { ChartProvider, ChartContext } from './ChartProvider'

export type Field = {
  fieldKey: string
  fieldName: string
  yAxisIndex?: number
  isPercent?: boolean
  decimalLength?: number
}

export type ChartProps = {
  dimension: Field[]
  valueList: Field[]
}

export { ChartProvider, ChartContext }
export default { ChartProvider, ChartContext }
