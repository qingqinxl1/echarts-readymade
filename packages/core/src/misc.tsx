import type { EChartsOption } from 'echarts-for-react'
import Big from 'big.js'

export const COLOR_LIST = [
  '#FF7C7C',
  '#ADD4EF',
  '#BFDAA7',
  '#FCAC65',
  '#C6C1D2',
  '#7598E4',
  '#CF6D6C',
  '#4979CF',
  '#E1934B',
  '#829649',
  '#7D70AC',
  '#2559B7'
]

export const echartsOptionsBase: any = {
  title: {
    show: true,
    text: '',
    top: 30,
    left: 24,
    textAlign: 'left',
    textStyle: {
      color: 'rgba(0,0,0,0.65)',
      fontSize: 16,
      lineHeight: 1
    }
  },
  grid: {
    x: 30,
    x2: 30,
    top: 90,
    bottom: 30,
    containLabel: true
  },
  toolbox: { show: false },
  tooltip: {
    show: true,
    trigger: 'axis',
    backgroundColor: 'rgba(255,255,255,0.9)',
    extraCssText:
      'box-shadow:  0px 0px 4px 0px rgba(0,0,0,0.25); border-radius: 2px; padding:10px 14px',
    formatter: function (data: any) {
      const _data = Array.isArray(data) ? data : [data]

      const _len = Big(_data.length)
      const _count = _len.div(10).round(0, Big.roundUp).toNumber()

      return `<div style="color: #c8c8c8;">${
        _data[0].name
      }</div><div style="column-count: ${_count};">${_data
        .map((d: any) => {
          switch (d.seriesType) {
            case 'pie':
              return `<div>
                      <span style="display:inline-block;margin-right:5px;border-radius:10px;width:9px;height:9px;background-color:${
                        d.color
                      }"></span>
                      <span style="color: #595959;">${d.seriesName}:</span>
                      <span style="color: #262626;">${
                        d.data.isShowPercentValue
                          ? d.percent + '%'
                          : d.data.value == null
                          ? '--'
                          : d.data.isPercent
                          ? d.data.value + '%'
                          : Big(d.data.value).toFixed(
                              d.data.decimalLength != null ? d.data.decimalLength : 2
                            )
                      }</span>
                  </div>`
            case 'line':
              return `<div>
                      <span style="display:inline-block;margin-right:5px;border-radius:10px;width:9px;height:9px;background-color:${
                        d.color
                      }"></span>
                      <span style="color: #595959;">${d.seriesName}:</span>
                      <span style="color: #262626;">${
                        d.data.value == null
                          ? '--'
                          : d.data.isPercent
                          ? d.data.value != null
                            ? `${d.data.value}%`
                            : `--`
                          : Big(d.data.value).toFixed(
                              d.data.decimalLength != null ? d.data.decimalLength : 2
                            )
                      }</span>
                  </div>`
            default:
              return `<div>
                      <span style="display:inline-block;margin-right:5px;border-radius:10px;width:9px;height:9px;background-color:${
                        d.color
                      }"></span>
                      <span style="color: #595959;">${d.seriesName}:</span>
                      <span style="color: #262626;">${
                        d.data.value == null
                          ? '--'
                          : d.data.isPercent
                          ? d.data.value + '%'
                          : Big(d.data.value).toFixed(
                              d.data.decimalLength != null ? d.data.decimalLength : 2
                            )
                      }</span>
                  </div>`
          }
        })
        .join('')}</div>`
    }
  },
  xAxis: {
    type: 'category',
    offset: 0,
    axisLine: {
      show: true,
      lineStyle: {
        color: '#D9D9D9'
      }
    },
    axisTick: {
      show: true,
      lineStyle: {}
    },
    splitLine: {
      show: false,
      lineStyle: {}
    },
    axisLabel: {
      rotate: 45,
      align: 'right',
      textStyle: {
        color: '#8C8C8C'
      }
    },
    data: []
  },
  yAxis: [
    {
      show: true,
      type: 'value',
      name: '',
      // scale: true,
      splitLine: {
        show: true,
        lineStyle: {
          color: '#E7EAEF',
          type: 'dashed'
        }
      },
      axisLine: {
        show: false,
        lineStyle: {
          color: '#D9D9D9'
        }
      },
      axisLabel: {
        textStyle: {
          color: '#8C8C8C'
        }
      },
      axisTick: {
        show: false,
        lineStyle: {}
      }
      // min: 0,
    },
    {
      show: false,
      type: 'value',
      name: '',
      // scale: true,
      splitLine: {
        show: false,
        lineStyle: {}
      },
      axisLine: {
        show: false,
        lineStyle: {
          color: '#D9D9D9'
        }
      },
      axisLabel: {
        textStyle: {
          color: '#8C8C8C'
        }
      },
      axisTick: {
        show: false,
        lineStyle: {}
      }
      // min: 0,
    }
  ],
  color: COLOR_LIST,
  textStyle: {},
  legend: {
    itemWidth: 10,
    itemHeight: 10,
    top: 30,
    left: 24,
    selectedMode: false,
    textStyle: {
      padding: [2, 0, 0, 0]
    },
    show: true,
    type: 'scroll',
    position: 'bottom',
    fontSize: 12,
    icon: 'circle'
  },
  series: null
}

export const mergeOption = (
  baseOptions: EChartsOption,
  userOptions: EChartsOption = {}
): EChartsOption => {
  const _baseOptions = baseOptions || {}
  const _userOptions = userOptions || {}
  const _mergeOptions = {
    ..._baseOptions,
    ..._userOptions
  }
  return _mergeOptions
}

export const truncate = (str: string, n: number) => {
  const r = /[^\x00-\xff]/g
  let m

  if (!str) {
    return ''
  }

  if (str.replace(r, '**').length > n) {
    m = Math.floor(n / 2)

    for (let i = m, l = str.length; i < l; i++) {
      if (str.slice(0, i).replace(r, '**').length >= n) {
        return str.slice(0, i) + '...'
      }
    }
  }

  return str
}

export function numberWithCommas(x: string | number) {
  if (!x) {
    return '--'
  }
  const parts = x.toString().split('.')
  parts[0] = parts[0].replace(/\B(?=(\d{3})+(?!\d))/g, ',')
  return parts.join('.')
}
