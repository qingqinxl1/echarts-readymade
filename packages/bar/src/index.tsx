import React, { useContext } from 'react'
import { multiply, round } from 'mathjs/number'
import { cloneDeep } from 'lodash'
import ReactEChartsCore from 'echarts-for-react/lib/core'
import * as echarts from 'echarts/core'
import type { ChartProps } from '@echarts-readymade/core'
import { mergeOption, buildChartOption } from '../../../packages/core/src'
import { ChartContext } from '../../../packages/core/src/ChartProvider'
import { BarChart } from 'echarts/charts'
import {
  GridSimpleComponent,
  GridComponent,
  SingleAxisComponent,
  GraphicComponent,
  ToolboxComponent,
  TooltipComponent,
  AxisPointerComponent,
  BrushComponent,
  TitleComponent,
  TimelineComponent,
  MarkPointComponent,
  MarkLineComponent,
  MarkAreaComponent,
  LegendComponent,
  LegendScrollComponent,
  LegendPlainComponent,
  DataZoomComponent,
  DataZoomInsideComponent,
  DataZoomSliderComponent,
  VisualMapComponent,
  VisualMapContinuousComponent,
  VisualMapPiecewiseComponent,
  AriaComponent,
  TransformComponent,
  DatasetComponent
} from 'echarts/components'

import { CanvasRenderer, SVGRenderer } from 'echarts/renderers'

echarts.use([
  BarChart,
  CanvasRenderer,
  SVGRenderer,
  GridSimpleComponent,
  GridComponent,
  SingleAxisComponent,
  GraphicComponent,
  ToolboxComponent,
  TooltipComponent,
  AxisPointerComponent,
  BrushComponent,
  TitleComponent,
  TimelineComponent,
  MarkPointComponent,
  MarkLineComponent,
  MarkAreaComponent,
  LegendComponent,
  LegendScrollComponent,
  LegendPlainComponent,
  DataZoomComponent,
  DataZoomInsideComponent,
  DataZoomSliderComponent,
  VisualMapComponent,
  VisualMapContinuousComponent,
  VisualMapPiecewiseComponent,
  AriaComponent,
  TransformComponent,
  DatasetComponent
])

export type LegendPosition = 'top' | 'left' | 'right' | 'bottom'

interface LineChartProps extends ChartProps {
  legendPosition?: LegendPosition
}

export const Bar: React.FC<LineChartProps> = (props) => {
  const {
    data,
    echartsOptions,
    echartsOptionsBase: chartOption,
    userOptions
  } = useContext(ChartContext)
  const { option, ...resetOptions } = echartsOptions || {}
  const { dimension, valueList, echartsSeries, xAxisData, ...restSettings } = props

  if (chartOption) {
    chartOption.xAxis.data =
      xAxisData ||
      (data &&
        data.map((d) => {
          const value = d[dimension && dimension[0] && dimension[0].fieldKey]
          if (value != null) {
            return `${value}`
          }
        }))
    chartOption.series =
      echartsSeries ||
      valueList.map((v) => {
        return {
          name: v.fieldName,
          type: 'bar',
          barMaxWidth: 60,
          barGap: 0,
          data:
            data &&
            data.map((d) => {
              if (d[v.fieldKey] != null) {
                let result = d[v.fieldKey]
                if (v.isPercent) {
                  result = multiply(d[v.fieldKey], 100)
                }
                return {
                  value: round(result, v.decimalLength || 0),
                  isPercent: v.isPercent
                }
              }
              return {
                value: 0,
                isPercent: v.isPercent
              }
            }),
          yAxisIndex: v.yAxisIndex || 0
        }
      })
  }

  const builtOption = buildChartOption(chartOption, restSettings, 'bar')
  const options = mergeOption(builtOption, userOptions)

  return (
    <>
      <ReactEChartsCore
        echarts={echarts}
        option={{ ...cloneDeep(options) }}
        notMerge={true}
        opts={{ renderer: 'svg' }}
        style={{ height: '100%', width: '100%' }}
        {...resetOptions}
      />
    </>
  )
}
