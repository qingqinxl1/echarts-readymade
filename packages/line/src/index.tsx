import React, { useContext, useState } from 'react'
import { multiply, round } from 'mathjs/number'
import { cloneDeep } from 'lodash'
import ReactEChartsCore from 'echarts-for-react/lib/core'
import * as echarts from 'echarts/core'
import type { ChartProps, LegendPosition } from '@echarts-readymade/core'
import { mergeOption, buildChartOption } from '../../../packages/core/src'
import { ChartContext } from '../../../packages/core/src/ChartProvider'
import { LineChart } from 'echarts/charts'
import {
  GridSimpleComponent,
  GridComponent,
  SingleAxisComponent,
  GraphicComponent,
  ToolboxComponent,
  TooltipComponent,
  AxisPointerComponent,
  BrushComponent,
  TitleComponent,
  TimelineComponent,
  MarkPointComponent,
  MarkLineComponent,
  MarkAreaComponent,
  LegendComponent,
  LegendScrollComponent,
  LegendPlainComponent,
  DataZoomComponent,
  DataZoomInsideComponent,
  DataZoomSliderComponent,
  VisualMapComponent,
  VisualMapContinuousComponent,
  VisualMapPiecewiseComponent,
  AriaComponent,
  TransformComponent,
  DatasetComponent
} from 'echarts/components'

import { CanvasRenderer, SVGRenderer } from 'echarts/renderers'

echarts.use([
  LineChart,
  CanvasRenderer,
  SVGRenderer,
  GridSimpleComponent,
  GridComponent,
  SingleAxisComponent,
  GraphicComponent,
  ToolboxComponent,
  TooltipComponent,
  AxisPointerComponent,
  BrushComponent,
  TitleComponent,
  TimelineComponent,
  MarkPointComponent,
  MarkLineComponent,
  MarkAreaComponent,
  LegendComponent,
  LegendScrollComponent,
  LegendPlainComponent,
  DataZoomComponent,
  DataZoomInsideComponent,
  DataZoomSliderComponent,
  VisualMapComponent,
  VisualMapContinuousComponent,
  VisualMapPiecewiseComponent,
  AriaComponent,
  TransformComponent,
  DatasetComponent
])

interface LineChartProps extends ChartProps {
  xAxisData?: any[]
  legendPosition?: LegendPosition
}

export const Line: React.FC<LineChartProps> = (props) => {
  const {
    data,
    echartsOptions,
    echartsOptionsBase: chartOption,
    userOptions
  } = useContext(ChartContext)
  const { option, ...resetOptions } = echartsOptions || {}
  const { dimension, compareDimension, valueList, echartsSeries, xAxisData, ...restSettings } =
    props

  const _dimension = dimension && dimension.slice(0, 1)
  const _chartOption = chartOption || {}

  if (_chartOption) {
    let _xAxis = []
    const _preProcessData: any[] = []
    let _processData = []
    let compareDimensionValues = []
    if (compareDimension && compareDimension.length > 0) {
      // 对比维度全量值
      compareDimensionValues = [
        ...new Set(
          data?.map((d) => {
            return compareDimension
              .map((dim) => {
                return d[dim.fieldKey]
              })
              .join('~')
          })
        )
      ]
      compareDimensionValues = compareDimensionValues.filter((cdv) => {
        return !(cdv.startsWith('~') || cdv.endsWith('~'))
      })

      // 按维度分组
      data?.forEach((d) => {
        const _index = _preProcessData.findIndex(
          (item) => item.name == (_dimension && d[_dimension?.[0].fieldKey])
        )
        if (_index != -1) {
          _preProcessData[_index].data.push(d)
        } else {
          _preProcessData.push({
            name: _dimension && d[_dimension?.[0].fieldKey],
            data: [d]
          })
        }
      })

      // 在维度分组基础上，分组对比维度
      _processData = compareDimensionValues.map((item) => {
        return {
          name: item,
          data: _preProcessData.map((pd) => {
            const v = pd.data.find(
              (d: any) =>
                compareDimension
                  .map((dim) => {
                    return d[dim.fieldKey]
                  })
                  .join('~') == item
            )

            if (v) {
              return v
            } else {
              const _v: { [key: string]: any } = {}
              if (_dimension && _dimension[0]) {
                _v[_dimension[0].fieldKey] = pd.name
              }
              return _v
            }
          })
        }
      })

      // X轴数据
      let _data = [
        ...new Set(
          data?.map((d) => {
            return _dimension && d[_dimension[0].fieldKey]
          })
        )
      ]

      _xAxis = [
        {
          type: 'category',
          splitLine: {
            show: false,
            lineStyle: {}
          },
          axisLine: {
            show: true,
            lineStyle: {}
          },
          axisTick: {
            show: true,
            lineStyle: {}
          },
          axisLabel: {
            padding: [0, 0, 0, 0],
            rotate: 30
          },
          data: _data
        }
      ]

      // Y轴数据
      let _seriesValueList: any[] = []
      for (let i = 0; i < _processData.length; i++) {
        let _data = cloneDeep(_processData[i].data) || []
        let compareDimensionName = `${_processData[i].name}`

        valueList?.forEach((v) => {
          _seriesValueList.push({
            name:
              valueList.length > 1
                ? `${compareDimensionName}~${v.fieldName}`
                : compareDimensionName,
            type: 'line',
            barGap: 0,
            barMaxWidth: 60,
            lineStyle: {
              shadowColor: 'rgba(0,0,0,0.15)',
              shadowBlur: 3,
              shadowOffsetX: 0,
              shadowOffsetY: 1
            },
            data: _data.map((d) => {
              if (d[v.fieldKey] != null) {
                let result = d[v.fieldKey]
                if (v.isPercent) {
                  result = multiply(d[v.fieldKey], 100)
                }
                return {
                  value: round(result, v.decimalLength || 0),
                  isPercent: v.isPercent
                }
              }
              return {
                value: 0,
                isPercent: v.isPercent
              }
            }),
            yAxisIndex: v.yAxisIndex || 0
          })
        })
      }
      _chartOption.xAxis = _xAxis
      _chartOption.series = _seriesValueList
    } else {
      // 无对比维度
      _chartOption.xAxis.data =
        xAxisData ||
        (data &&
          data.map((d) => {
            const value = dimension && d[dimension?.[0]?.fieldKey]
            if (value != null) {
              return `${value}`
            }
          }))
      _chartOption.series =
        echartsSeries ||
        valueList?.map((v) => {
          return {
            name: v.fieldName,
            type: 'line',
            lineStyle: {
              shadowColor: 'rgba(0,0,0,0.15)',
              shadowBlur: 3,
              shadowOffsetX: 0,
              shadowOffsetY: 1
            },
            data:
              data &&
              data.map((d) => {
                if (d[v.fieldKey] != null) {
                  let result = d[v.fieldKey]
                  if (v.isPercent) {
                    result = multiply(d[v.fieldKey], 100)
                  }
                  return {
                    value: round(result, v.decimalLength || 0),
                    isPercent: v.isPercent
                  }
                }
                return {
                  value: 0,
                  isPercent: v.isPercent
                }
              }),
            yAxisIndex: v.yAxisIndex || 0
          }
        })
    }
  }

  const builtOption = buildChartOption(_chartOption, restSettings, 'line')
  const options = mergeOption(builtOption, userOptions)

  return (
    <>
      <ReactEChartsCore
        echarts={echarts}
        option={{ ...cloneDeep(options) }}
        notMerge={true}
        opts={{ renderer: 'svg' }}
        style={{ height: '100%', width: '100%' }}
        {...resetOptions}
      />
    </>
  )
}
