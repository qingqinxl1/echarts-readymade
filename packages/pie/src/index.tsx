import React, { useContext } from 'react'
import { multiply, round, divide, fix } from 'mathjs/number'
import { cloneDeep } from 'lodash'
import ReactEChartsCore from 'echarts-for-react/lib/core'
import * as echarts from 'echarts/core'
import type { ChartProps, LegendPosition } from '@echarts-readymade/core'
import { mergeOption, buildChartOption } from '../../../packages/core/src'
import { ChartContext } from '../../../packages/core/src/ChartProvider'
import { PieChart } from 'echarts/charts'
import {
  GridSimpleComponent,
  GridComponent,
  SingleAxisComponent,
  GraphicComponent,
  ToolboxComponent,
  TooltipComponent,
  AxisPointerComponent,
  BrushComponent,
  TitleComponent,
  TimelineComponent,
  MarkPointComponent,
  MarkLineComponent,
  MarkAreaComponent,
  LegendComponent,
  LegendScrollComponent,
  LegendPlainComponent,
  DataZoomComponent,
  DataZoomInsideComponent,
  DataZoomSliderComponent,
  VisualMapComponent,
  VisualMapContinuousComponent,
  VisualMapPiecewiseComponent,
  AriaComponent,
  TransformComponent,
  DatasetComponent
} from 'echarts/components'

import { CanvasRenderer, SVGRenderer } from 'echarts/renderers'

echarts.use([
  PieChart,
  CanvasRenderer,
  SVGRenderer,
  GridSimpleComponent,
  GridComponent,
  SingleAxisComponent,
  GraphicComponent,
  ToolboxComponent,
  TooltipComponent,
  AxisPointerComponent,
  BrushComponent,
  TitleComponent,
  TimelineComponent,
  MarkPointComponent,
  MarkLineComponent,
  MarkAreaComponent,
  LegendComponent,
  LegendScrollComponent,
  LegendPlainComponent,
  DataZoomComponent,
  DataZoomInsideComponent,
  DataZoomSliderComponent,
  VisualMapComponent,
  VisualMapContinuousComponent,
  VisualMapPiecewiseComponent,
  AriaComponent,
  TransformComponent,
  DatasetComponent
])

export interface PieChartProps extends ChartProps {
  showInRing?: boolean
  legendPosition?: LegendPosition
}

export const Pie: React.FC<PieChartProps> = (props) => {
  const {
    data,
    echartsOptions,
    echartsOptionsBase: chartOption,
    userOptions
  } = useContext(ChartContext)
  const { option, ...resetOptions } = echartsOptions || {}
  const { dimension, valueList, echartsSeries, showInRing, ...restSettings } = props

  const getCenter = () => {
    let _centerLeft = '50%'
    let _centerTop = '50%'
    if (props.legendPosition === 'left') {
      _centerLeft = '55%'
    } else if (props.legendPosition === 'right') {
      _centerLeft = '45%'
    } else if (props.legendPosition === 'top') {
      _centerTop = '55%'
    } else if (props.legendPosition === 'bottom') {
      _centerTop = '45%'
    }

    return {
      top: _centerTop,
      left: _centerLeft
    }
  }

  const _chartOption = Object.assign({}, chartOption || {}, {})

  _chartOption.xAxis = { show: false }
  _chartOption.yAxis = { show: false }
  let _series: any[] = []

  if (_chartOption) {
    if (dimension && dimension.length > 0) {
      // 有维度
      _series =
        (valueList &&
          valueList.slice(0, 1).map((item, i) => {
            const _sum = data?.reduce((s, c) => {
              s = s + (c[item.fieldKey] || 0)
              return s
            }, 0)

            return {
              name: dimension?.[0].fieldName,
              type: 'pie',
              minAngle: 5,
              radius: showInRing ? ['30%', '50%'] : '50%',
              center: [getCenter().left, getCenter().top],
              data: data?.map((d) => {
                let text = d[dimension?.[0].fieldKey]

                return {
                  name: text,
                  value: d[item.fieldKey]
                    ? d[item.fieldKey] != null && !isNaN(d[item.fieldKey])
                      ? d[item.fieldKey].toFixed(Math.abs(d[item.fieldKey]) > 1 ? 2 : 4)
                      : d[item.fieldKey]
                    : 0
                }
              }),
              label: {
                show: true,
                position: 'outside',
                formatter: (params: any) => {
                  const _percent = round(
                    multiply(divide(round(Number(params.value), 4), _sum), 100),
                    2
                  )

                  let res = ''
                  try {
                    res += `${_percent}%`
                  } catch {
                    res += '--%'
                  }
                  return res
                },
                fontSize: 12
              }
            }
          })) ||
        []
    } else {
      // 无维度
      const d = (data && cloneDeep(data)[0]) || {}
      const _valueListClone = cloneDeep(valueList)

      const _sum = _valueListClone?.reduce((s, c) => {
        s = s + (d[c.fieldKey] || 0)
        return s
      }, 0)

      _series = [
        {
          name: '',
          type: 'pie',
          minAngle: 5,
          radius: showInRing ? ['30%', '50%'] : '50%',
          center: [getCenter().left, getCenter().top],
          data: _valueListClone?.map((item) => {
            return {
              name: item.fieldName,
              value: d[item.fieldKey]
                ? d[item.fieldKey] != null && !isNaN(d[item.fieldKey])
                  ? parseFloat(d[item.fieldKey]).toFixed(Math.abs(d[item.fieldKey]) > 1 ? 2 : 4)
                  : d[item.fieldKey]
                : 0,
              emphasis: {
                label: {
                  show: true
                }
              },
              label: {
                show: true,
                position: 'outside',
                formatter: (params: any) => {
                  try {
                    const _percent = round(
                      multiply(divide(round(Number(params.value), 4), _sum), 100),
                      2
                    )

                    let res = ''
                    try {
                      res += `${_percent}%`
                    } catch {
                      res += '--%'
                    }
                    return res
                  } catch (e) {
                    return '--'
                  }
                },
                fontSize: 12
              }
            }
          })
        }
      ]
    }
  }

  _chartOption.series = echartsSeries || _series
  const builtOption = buildChartOption(_chartOption, restSettings, 'pie')
  const options = mergeOption(builtOption, userOptions)

  return (
    <>
      <ReactEChartsCore
        echarts={echarts}
        option={{ ...cloneDeep(options) }}
        notMerge={true}
        opts={{ renderer: 'svg' }}
        style={{ height: '100%', width: '100%' }}
        {...resetOptions}
      />
    </>
  )
}
