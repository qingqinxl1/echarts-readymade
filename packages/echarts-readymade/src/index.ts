export * from '@echarts-readymade/core'
import { Bar } from './bar'
import { Line } from './line'
import { Pie } from './pie'
import { Scatter } from './scatter'
import { Stack } from './stack'
import { Wordcloud } from './wordcloud'
import { BarHorizontal } from './bar-horizontal'
import { Table } from './table'
import type { IWordcloudChartProps } from './wordcloud'
import type { ITableChartProps } from './table'
import type { IStackChartProps } from './stack'
import type { IScatterChartProps } from './scatter'
import type { IPieChartProps } from './pie'
import type { ILineChartProps } from './line'
import type { IBarChartProps } from './bar'
import type { IBarHorizontalChartProps } from './bar-horizontal'

export type {
  IWordcloudChartProps,
  ITableChartProps,
  IStackChartProps,
  IScatterChartProps,
  IPieChartProps,
  ILineChartProps,
  IBarChartProps,
  IBarHorizontalChartProps
}
export { Bar, Line, Pie, Scatter, Stack, Wordcloud, BarHorizontal, Table }
export default { Bar, Line, Pie, Scatter, Stack, Wordcloud, BarHorizontal, Table }
