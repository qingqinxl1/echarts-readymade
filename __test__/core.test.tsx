import '@testing-library/jest-dom'
import { render, waitFor, screen, act } from '@testing-library/react'
import React from 'react'
import { buildChartOption } from '../packages/core/src/index'
import { echartsOptionsBase } from '../packages/core/src/misc'

describe('testing buildChartOption()', () => {
  it('buildChartOption() work fine', async () => {
    const option = buildChartOption(echartsOptionsBase, { legendPosition: 'top' }, 'bar')
    expect(option).toMatchSnapshot('buildChartOption() work fine 1')
  })
})
