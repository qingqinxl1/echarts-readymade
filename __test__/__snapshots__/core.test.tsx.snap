// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing buildChartOption() buildChartOption() work fine: buildChartOption() work fine 1 1`] = `
Object {
  "color": Array [
    "#FF7C7C",
    "#ADD4EF",
    "#BFDAA7",
    "#FCAC65",
    "#C6C1D2",
    "#7598E4",
    "#CF6D6C",
    "#4979CF",
    "#E1934B",
    "#829649",
    "#7D70AC",
    "#2559B7",
  ],
  "grid": Object {
    "bottom": 45,
    "containLabel": true,
    "left": 20,
    "right": 20,
    "top": 110,
    "x": 30,
    "x2": 30,
  },
  "legend": Object {
    "bottom": "auto",
    "data": undefined,
    "fontSize": 12,
    "formatter": [Function],
    "icon": "circle",
    "itemHeight": 10,
    "itemWidth": 10,
    "left": "center",
    "orient": "horizontal",
    "padding": Array [
      5,
      5,
    ],
    "position": "bottom",
    "selectedMode": false,
    "show": true,
    "textStyle": Object {
      "padding": Array [
        2,
        0,
        0,
        0,
      ],
    },
    "top": 60,
    "type": "scroll",
  },
  "series": null,
  "textStyle": Object {},
  "title": Object {
    "left": 24,
    "show": true,
    "text": "",
    "textAlign": "left",
    "textStyle": Object {
      "color": "rgba(0,0,0,0.65)",
      "fontSize": 16,
      "lineHeight": 1,
    },
    "top": 30,
  },
  "toolbox": Object {
    "show": false,
  },
  "tooltip": Object {
    "backgroundColor": "rgba(255,255,255,0.9)",
    "extraCssText": "box-shadow:  0px 0px 4px 0px rgba(0,0,0,0.25); border-radius: 2px; padding:10px 14px",
    "formatter": [Function],
    "show": true,
    "trigger": "axis",
  },
  "xAxis": Object {
    "axisLabel": Object {
      "align": "right",
      "rotate": 45,
      "textStyle": Object {
        "color": "#8C8C8C",
      },
    },
    "axisLine": Object {
      "lineStyle": Object {
        "color": "#D9D9D9",
      },
      "show": true,
    },
    "axisTick": Object {
      "lineStyle": Object {},
      "show": true,
    },
    "data": Array [],
    "offset": 0,
    "splitLine": Object {
      "lineStyle": Object {},
      "show": false,
    },
    "type": "category",
  },
  "yAxis": Array [
    Object {
      "axisLabel": Object {
        "textStyle": Object {
          "color": "#8C8C8C",
        },
      },
      "axisLine": Object {
        "lineStyle": Object {
          "color": "#D9D9D9",
        },
        "show": false,
      },
      "axisTick": Object {
        "lineStyle": Object {},
        "show": false,
      },
      "name": "",
      "show": true,
      "splitLine": Object {
        "lineStyle": Object {
          "color": "#E7EAEF",
          "type": "dashed",
        },
        "show": true,
      },
      "type": "value",
    },
    Object {
      "axisLabel": Object {
        "textStyle": Object {
          "color": "#8C8C8C",
        },
      },
      "axisLine": Object {
        "lineStyle": Object {
          "color": "#D9D9D9",
        },
        "show": false,
      },
      "axisTick": Object {
        "lineStyle": Object {},
        "show": false,
      },
      "name": "",
      "show": false,
      "splitLine": Object {
        "lineStyle": Object {},
        "show": false,
      },
      "type": "value",
    },
  ],
}
`;

exports[`testing buildChartOption() buildChartOption() works fine: buildChartOption() work fine 1 1`] = `
Object {
  "color": Array [
    "#FF7C7C",
    "#ADD4EF",
    "#BFDAA7",
    "#FCAC65",
    "#C6C1D2",
    "#7598E4",
    "#CF6D6C",
    "#4979CF",
    "#E1934B",
    "#829649",
    "#7D70AC",
    "#2559B7",
  ],
  "grid": Object {
    "bottom": 45,
    "containLabel": true,
    "left": 20,
    "right": 20,
    "top": 110,
    "x": 30,
    "x2": 30,
  },
  "legend": Object {
    "bottom": "auto",
    "data": undefined,
    "fontSize": 12,
    "formatter": [Function],
    "icon": "circle",
    "itemHeight": 10,
    "itemWidth": 10,
    "left": "center",
    "orient": "horizontal",
    "padding": Array [
      5,
      5,
    ],
    "position": "bottom",
    "selectedMode": false,
    "show": true,
    "textStyle": Object {
      "padding": Array [
        2,
        0,
        0,
        0,
      ],
    },
    "top": 60,
    "type": "scroll",
  },
  "series": null,
  "textStyle": Object {},
  "title": Object {
    "left": 24,
    "show": true,
    "text": "",
    "textAlign": "left",
    "textStyle": Object {
      "color": "rgba(0,0,0,0.65)",
      "fontSize": 16,
      "lineHeight": 1,
    },
    "top": 30,
  },
  "toolbox": Object {
    "show": false,
  },
  "tooltip": Object {
    "backgroundColor": "rgba(255,255,255,0.9)",
    "extraCssText": "box-shadow:  0px 0px 4px 0px rgba(0,0,0,0.25); border-radius: 2px; padding:10px 14px",
    "formatter": [Function],
    "show": true,
    "trigger": "axis",
  },
  "xAxis": Object {
    "axisLabel": Object {
      "align": "right",
      "rotate": 45,
      "textStyle": Object {
        "color": "#8C8C8C",
      },
    },
    "axisLine": Object {
      "lineStyle": Object {
        "color": "#D9D9D9",
      },
      "show": true,
    },
    "axisTick": Object {
      "lineStyle": Object {},
      "show": true,
    },
    "data": Array [],
    "offset": 0,
    "splitLine": Object {
      "lineStyle": Object {},
      "show": false,
    },
    "type": "category",
  },
  "yAxis": Array [
    Object {
      "axisLabel": Object {
        "textStyle": Object {
          "color": "#8C8C8C",
        },
      },
      "axisLine": Object {
        "lineStyle": Object {
          "color": "#D9D9D9",
        },
        "show": false,
      },
      "axisTick": Object {
        "lineStyle": Object {},
        "show": false,
      },
      "name": "",
      "show": true,
      "splitLine": Object {
        "lineStyle": Object {
          "color": "#E7EAEF",
          "type": "dashed",
        },
        "show": true,
      },
      "type": "value",
    },
    Object {
      "axisLabel": Object {
        "textStyle": Object {
          "color": "#8C8C8C",
        },
      },
      "axisLine": Object {
        "lineStyle": Object {
          "color": "#D9D9D9",
        },
        "show": false,
      },
      "axisTick": Object {
        "lineStyle": Object {},
        "show": false,
      },
      "name": "",
      "show": false,
      "splitLine": Object {
        "lineStyle": Object {},
        "show": false,
      },
      "type": "value",
    },
  ],
}
`;
